client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -186,21 +187,25 @@ private Checkpoint completeCheckpoint(String checkpointName) {
     public void resetReadersToCheckpoint(Checkpoint checkpoint) {
         synchronizer.updateState((state, updates) -> {
             ReaderGroupConfig config = state.getConfig();
-            Map<Segment, Long> positions = new HashMap<>();
+            Map<SegmentWithRange, Long> positions = new HashMap<>();
             for (StreamCut cut : checkpoint.asImpl().getPositions().values()) {
-                positions.putAll(cut.asImpl().getPositions());
+                for (Entry<Segment, Long> e : cut.asImpl().getPositions().entrySet()) {
+                    //TODO watermarking: deal with empty range information.
```

Comment:
we should create a tracking issue for this.

Reply:
https://github.com/pravega/pravega/issues/4194